engine/src/main/scala/cromwell/engine/workflow/lifecycle/execution/stores/ExecutionStore.scala

```diff
@@ -41,6 +41,12 @@ object ExecutionStore {
         case _ => key.node.upstreamPorts forall { p => p.executionNode.isInStatus(chooseIndex(p), statusTable) }
       }
     }
+    
+    def nonStartableOutputKeys: Set[JobKey] = key match {
+      case scatterKey: ScatterKey => scatterKey.makeCollectors(0, scatterKey.node.scatterCollectionFunctionBuilder(List.empty)).toSet[JobKey]
```

Comment:
ToL: Do we have to recreate the entire set of collectors every time we run this `def nonStartableOutputKeys`?

Reply:
This is really going to *maybe* be called only once per key per workflow anyway. Since it's in an implicit class it has to be a `def`.